std/random.d

```diff
@@ -2634,6 +2634,104 @@ do
     assert(i == 0);
 }
 
+/+ @nogc bool array designed for RandomCover.
+- constructed with an invariable length
+- small length means 0 alloc and bit field (if up to 32 and 64 elements to cover)
+- bigger length means non-GC heap allocation(s) and dealloc. +/
+private struct RandomCoverChoices
+{
+    private static immutable emsg = "No memory available to store random cover choices";
+    private void* buffer;
+    private immutable size_t _length;
+    private immutable bool hasPackedBits;
+
+    this(this) nothrow @nogc @trusted
+    {
+        import core.stdc.stdlib : malloc;
+        import core.stdc.string : memmove;
+
+        if (!hasPackedBits && buffer !is null)
+        {
+            void* nbuffer = malloc(_length);
+            if (nbuffer is null)
+                assert(0, emsg);
```

Comment:
In places like this it seems like the convention is to use [`onOutOfMemoryError`](https://dlang.org/phobos/core_exception.html#.onOutOfMemoryError):
```d
if (nbuffer is null)
{
    onOutOfMemoryError();
    assert(0);
}
```
...but this has the downside that your custom message would be lost.